pipeline.libsonnet

```diff
@@ -0,0 +1,187 @@
+local windows_pipe = '\\\\\\\\.\\\\pipe\\\\docker_engine';
+local windows_pipe_volume = 'docker_pipe';
+local test_pipeline_name = 'testing';
+
+local windows(os) = os == 'windows';
+
+local golang_image(os, version) =
+  'golang:' + '1.11' + if windows(os) then '-windowsservercore-' + version else '';
+
+{
+  test(os='linux', arch='amd64', version='')::
+    local golang = golang_image(os, version);
+    {
+      kind: 'pipeline',
+      name: test_pipeline_name,
+      platform: {
+        os: os,
+        arch: arch,
+        version: if std.length(version) > 0 then version,
+      },
+      steps: [
+        {
+          name: 'vet',
+          image: golang,
+          pull: 'always',
+          environment: {
+            GO111MODULE: 'on',
+          },
+          commands: [
+            'go vet ./...',
+          ],
+        },
+        {
+          name: 'test',
+          image: golang,
+          pull: 'always',
+          environment: {
+            GO111MODULE: 'on',
+          },
+          commands: [
+            'go test -cover ./...',
+          ],
+        },
+      ],
+    },
+
+  build(name, os='linux', arch='amd64', version='')::
+    local is_windows = windows(os);
+    local tag = if is_windows then os + '-' + version else os + '-' + arch;
+    local file_suffix = std.strReplace(tag, '-', '.');
+    local volumes = if is_windows then [{ name: windows_pipe_volume, path: windows_pipe }] else [];
+    local golang = golang_image(os, version);
+    local plugin_repo = 'plugins/' + std.split(name, '-')[1];
```

Comment:
We could just supply the name `webhook`, then we don't need to split it. Besides that, this logic will break for plugins like `drone-volume-cache`.

Reply:
std.splitLimit(str, c, maxsplits) does the trick.